python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -247,7 +323,29 @@ def modelToHeavy(source, filters, xy0=Point2I(), observation=None, dtype=np.floa
         The multi-band footprint containing the model for the source.
     """
     if observation is not None:
-        model = observation.render(source.get_model()).astype(dtype)
+        # We want to convolve the model with the observed PSF,
+        # which means we need to grow the model box by the PSF to
+        # account for all of the flux after convolution.
+        # FYI: The `scarlet.Box` class implements the `&` operator
+        # to take the intersection of two boxes.
+
+        # Get the PSF size and radii to grow the
+        py, px = observation.frame.psf.shape[1:]
+        dh = py // 2
+        dw = px // 2
+        shape = (source.bbox.shape[0], source.bbox.shape[1] + py, source.bbox.shape[2] + px)
+        origin = (source.bbox.origin[0], source.bbox.origin[1] - dh, source.bbox.origin[2] - dw)
+        # Create the larger box to fit the model + PSf
+        bbox = Box(shape, origin=origin)
+        # Only use the portion of the convolved model that fits in the image
+        overlap = bbox & source.model_frame
+        # Load the full multiband model in the larger box
+        model = source.model_to_frame(overlap)
+        # Convolve the model with the PSF in each band
+        # Always use a real space convolution to limit artifacts
+        model = observation.convolve(model, convolution_type="real").astype(dtype)
+        # Update xy0 with the origin of the sources box
+        xy0 = Point2I(overlap.origin[-1]+xy0.x, overlap.origin[-2]+xy0.y)
```

Comment:
The style guide says + and - should have spaces, so I'd suggest staying consistent with that (there are spaces just above).